module3/preparation/Week1/HTMLFundamentals.md

```diff
@@ -0,0 +1,44 @@
+---
+layout: page
+title: HTML Fundamentals Pre Work
+---
+
+Debugging your front-end code can be an intimidating part of the development process, but it’s also one of the most powerful skills you can acquire. Developers of all levels spend a significant amount of time troubleshooting code, but the more comfortable you are with debugging tools, the easier it will be to isolate, identify and fix broken code.
+
+The front-end languages (HTML, CSS and JavaScript) are run entirely in the browser, so the technique for troubleshooting broken code can happen in many places. Luckily, modern browsers are aware of this and give us a collection of advanced tools to help us debug.
+
+## Developer Tools
+
+One of the first tools you should familiarize yourself with when doing front-end development are the built-in browser Developer Tools, commonly referred to as "dev tools". To open in Chrome:
+
+* `Ctrl + Shift + i`
+* (or) Right click on the browser window and select inspect
+
+### The Elements Panel: Debugging HTML, CSS & DOM Events
+
+Most helpful for:
+- debugging layout and styling issues
+- monitoring DOM Events
+
+The elements panel lets you view the entire HTML source of the current page you are viewing. From here, you can edit, add or remove content and elements directly on the page. Though your changes won’t be saved (any changes made here will be lost upon refreshing the page), sometimes it’s helpful to make tweaks directly in this panel so you can see what effect the changes will have on your application before you implement them.
+
+## Dev Tools Introduction
+
+Please watch the video below, the complete the exercises to work towards that muscle memory of using the dev tools!
+
+<div style="position: relative; padding-bottom: 66.66666666666666%; height: 0;"><iframe src="https://www.loom.com/embed/b0b7d7935fd9494a915b7093a746f2b0" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>
+
+## Dev Tools Practice Exercises
+
+- Open the dev tools on your favorite site
+- Move the dock to the bottom
+- Close the dev tools
+- Re-open the dev tools - where is the dock?
+- Click the inspect arrow and click on something in the browser. What HTML element is it?
+- Click on something else in the browser, something you suspect will be a different HTML element.
+- Move the dock to the right
+- Close the dev tools
+- Re-open the dev tools - where is the dock?
+- Open the same site in another tab - are the dev tools open in that tab?
+- Close your dev tools
+- Open and close them **one last time** to make sure you're getting that keyboard shortcut down!
```

Comment:
What do you think about also having part of the practice be changing some text one one of the pages, like you showed in the video?

I'm also thinking about us getting in the habit of having a preparation "deliverable" that they will share during the lesson. And making that very clear, maybe in it's own section. I think a fun deliverable for this would be a screenshot of what the website looks like after you changed the text to something different.

Reply:
Yes, I love this deliverable idea!